internal/action/list.go

```diff
@@ -50,28 +51,33 @@ func (s *Action) List(c *cli.Context) error {
 		limit = c.Int("limit")
 	}
 
-	if filter == "" {
-		return s.listAll(ctx, l, limit, flat, folders)
-	}
-
 	return s.listFiltered(ctx, l, limit, flat, folders, stripPrefix, filter)
 }
 
 func (s *Action) listFiltered(ctx context.Context, l *tree.Root, limit int, flat, folders, stripPrefix bool, filter string) error {
-	subtree, err := l.FindFolder(filter)
-	if err != nil {
-		return ExitError(ExitNotFound, nil, "Entry '%s' not found", filter)
+
+	sep := string(filepath.Separator)
+
+	if filter == "" {
+		// We list all entries then
+		stripPrefix = true
+	} else {
+		// We restrict ourselves to the filter
+		var err error // to avoid shadowing l
```

Comment:
nit: Could you move this below the if statement, please?

Reply:
Also maybe extend the comment, e.g. "avoid shadowing l, we need it's value after the if clause" (or similar)